src/jit/codegenxarch.cpp

```diff
@@ -6706,30 +6718,46 @@ void         CodeGen::genJumpKindsForTreeLongHi(GenTreePtr    cmpTree,
     jmpToTrueLabel[1] = true;
 
     assert(cmpTree->OperIsCompare());
+    CompareKind compareKind = ((cmpTree->gtFlags & GTF_UNSIGNED) != 0) ? CK_UNSIGNED : CK_SIGNED;
 
```

Comment:
Removed. Also, since genJumpKindsForTreeLongLo basically only calls one other function, I have removed it, and replaced it with genJumpKindsForTree(tree->gtOper, CK_UNSIGNED).